lib/GLPI/Agent/Task/Inventory/Generic/Remote_Mgmt/SupRemo.pm

```diff
@@ -0,0 +1,129 @@
+            path => "HKEY_LOCAL_MACHINE/SOFTWARE/Wow6432Node/Supremo",
+            # Important for remote inventory optimization
+            required    => [ qw/ClientID/ ],
+            maxdepth    => 1,
+            logger => $params{logger}
+        );
+        if (!$key && is64bit()) {
+            $key = getRegistryKey(
+                path => "HKEY_LOCAL_MACHINE/SOFTWARE/Wow6432Node/Supremo",
+                # Important for remote inventory optimization
+                required    => [ qw/ClientID/ ],
+                maxdepth    => 1,
+                logger => $params{logger}
+            );
+        }
+
+        return $key && (keys %$key);
+    }
+
+    return canRun('supremo');
+}
+
+sub doInventory {
+    my (%params) = @_;
+
+    my $inventory = $params{inventory};
+    my $logger    = $params{logger};
+
+    my $supRemoID = _getID(
+        osname  => OSNAME,
+        logger  => $logger
+    );
+    if (defined($supRemoID)) {
+        $logger->debug('Found SupRemoID : ' . $supRemoID) if ($logger);
+
+        $inventory->addEntry(
+            section => 'REMOTE_MGMT',
+            entry   => {
+                ID   => $supRemoID,
+                TYPE => 'supremo'
+            }
+        );
+    } else {
+        $logger->debug('SupRemoID not found') if ($logger);
+    }
+}
+
+sub _getID {
+    my (%params) = @_;
+
+    my $osname = delete $params{osname} // '';
+
+    return _getID_MSWin32() if $osname eq "MSWin32";
+
+    return _getID_supremo_info(%params);
+}
+
+sub _getID_MSWin32 {
+
+    GLPI::Agent::Tools::Win32->use();
+
+    my $clientid = getRegistryValue(
+        path => "HKEY_LOCAL_MACHINE/SOFTWARE/Supremo/Wow6432Node/ClientID",
+    );
+    if (!$clientid && is64bit()) {
+        $clientid = getRegistryValue(
+            path => "HKEY_LOCAL_MACHINE/SOFTWARE/Wow6432Node/Supremo/ClientID",
+        );
+    }
+
+    unless ($clientid) {
+        my $supremover_reg = getRegistryKey(
+            path => "HKEY_LOCAL_MACHINE/SOFTWARE/Wow6432Node/Supremo",
+            # Important for remote inventory optimization
+            required    => [ qw/ClientID/ ],
+        );
+        if(!$supremover_reg && is64bit()){
+            $supremover_reg = getRegistryKey(
+                path => "HKEY_LOCAL_MACHINE/SOFTWARE/Wow6432Node/Supremo",
+                # Important for remote inventory optimization
+                required    => [ qw/ClientID/ ],
+            );
+        }
+
+        return unless $supremover_reg;
+
+        # Look for subkey beginning with Version
+        foreach my $key (keys(%{$supremover_reg})) {
+            next unless $key =~ /^Version\d+\//;
+            $clientid = $supremover_reg->{$key}->{"/ClientID"};
+            last if (defined($clientid));
+        }
+    }
+
+    return sprintf("%09d",hex2dec($clientid));
+}
+
+sub _getID_supremo_info {
+    my (%params) = @_;
+
+    return getFirstMatch(
+        command => "supremo --info",
+        pattern => qr/SupRemo ID:(?:\033\[0m|\s)*(\d+)\s+/,
```

Comment:
To avoid special chars in the regexp, can you check if there's an option for `supremo` command to have a machine readable output or just escape the pretty formatting ?

Reply:
There is no option from the executable to have a readable output